<!-- Modal TOOLS -->
<div class="modal fade" id="modal-lg" tabindex="-1" role="dialog" aria-labelledby="myModalLabel" aria-hidden="true">
  <div class="modal-dialog modal-lg">
	<div class="modal-content">
	  <div class="modal-header">
		<button type="button" class="close" data-dismiss="modal" aria-label="Close"><span aria-hidden="true">×</span></button>
		<!-- Titre -->
		<h4 class="modal-title">Integrated Bioinformatics Tools</h4>
	  </div>
	  <div class="modal-body">
<!--		<div class="alert alert-info">
		  TOGGLE currently supports bioinformatics tools, listed below.
		  If you would like another to be added, please <a target="_blank" href="https://github.com/SouthGreenPlatform/TOGGLE/issues" class="alert-link">open an issue</a>.
		</div>-->
		<ul class="list-inline mod-list-list">
			<!-- Catégories -->
		  <li class="mod-list datapreprocessing">Data preprocessing</li>
		  <li class="mod-list dataQC">Data QC</li>
		  <li class="mod-list demultiplex">Demultiplexing</li>
		  <li class="mod-list mappings">Mapping</li>
		 <li class="mod-list assembly">Assembly</li>
		 <li class="mod-list post-mappings">SAM/BAM</li>
		  <li class="mod-list vcf-tools">VCF</li>
		  <li class="mod-list rna-seq">RNA-seq</li>
		  <li class="mod-list struc-variants">Structural Variants</li>
		  <li class="mod-list optional">Optional</li>
		</ul>
		<p class="text-muted"><em><span class="fa fa-book"></span> &nbsp;Click the tool name to go to website of tools.</em></p>
		<table class="table table-condensed">
		  <tbody><tr>
			  <!-- header tableau -->
			<th>Software Name</th>
			<th>Minimal version</th>
			<th>Tools included</th>
		  </tr>
		  </tbody>
			<!-- TOOLS data QC -->
			 <tr class="mod-list dataQC">
				<td> <a target="_blank" href="/manual/internalTools/">fastqUtils</a></td>
				<td> 0.1 </td>
				<td> checkEncodeByASCIIcontrol </td>
			 </tr>
			 <tr class="mod-list dataQC">
				<td> <a target="_blank" href="/manual/internalTools/">checkFormat</a></td>
				<td> 0.1 </td>
				<td> checkFormatFastq <br/> checkFormatFasta <br/> checkFormatSamOrBam <br/> checkFormatVcf</td>
			 </tr>
			 <tr class="mod-list dataQC">
				<td> <a target="_blank" href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a></td>
				<td> 0.10.1 </td>
				<td> fastqc </td>
			 </tr>
			 <!-- TOOLS datapreprocessing -->
			 <tr class="mod-list datapreprocessing">
				<td> <a target="_blank" href="https://github.com/jdidion/atropos">Atropos</a></td>
				<td> 1.1.14 </td>
				<td> Atropos </td>
			  </tr>			 <tr class="mod-list datapreprocessing">
				<td> <a target="_blank" href="https://pypi.python.org/pypi/cutadapt">Cutadapt</a></td>
				<td> 1.2.1 </td>
				<td> cutadapt </td>
			  </tr>
			 <tr class="mod-list datapreprocessing">
				<td> <a target="_blank" href="http://hannonlab.cshl.edu/fastx_toolkit/">FastxToolkit</a></td>
				<td> 0.0.13 </td>
				<td> fastxTrimmer </td>
			 </tr>
			 <!-- TOOLS demultiplex -->
			 <tr class="mod-list demultiplex">
				<td> <a target="_blank" href="http://catchenlab.life.illinois.edu/stacks/">Stacks</a></td>
				<td> 1.43 </td>
				<td> process_radtags </td>
			 </tr>
		  <!-- TOOLS mappings -->
		  <tr class="mod-list mappings">
				<td> <a target="_blank" href="http://crac.gforge.inria.fr/">Crac</a></td>
				<td> 2.5.0 </td>
				<td> crac-index <br/> crac </td>
		  </tr>
		  <tr class="mod-list mappings">
				<td> <a target="_blank" href="http://bowtie-bio.sourceforge.net/manual.shtml">Bowtie</a></td>
				<td> 1.1.2 </td>
				<td> bowtiebuild <br/> bowtie </td>
		  </tr>
		  <tr class="mod-list mappings">
				<td> <a target="_blank" href="http://bowtie-bio.sourceforge.net/bowtie2/index.shtml">Bowtie2</a></td>
				<td> 2.2.9 </td>
				<td> bowtie2build <br/> bowtie2 </td>
		  </tr>
		  <tr class="mod-list mappings">
				<td> <a target="_blank" href="http://bio-bwa.sourceforge.net/">BWA</a></td>
				<td> 0.7.2 </td>
				<td> bwaAln <br/> bwaSampe <br/> bwaSamse <br/> bwaMem <br/>bwaSW<br/> bwaIndex</td>
		  </tr>
		  <tr class="mod-list mappings">
				<td> <a target="_blank" href="https://ccb.jhu.edu/software/tophat/index.shtml">Tophat</a></td>
				<td> 2.1.1 </td>
				<td> tophat2 </td>
		  </tr>
			<!-- TOOLS assembly -->
		  <tr class="mod-list assembly">
				<td> <a target="_blank" href="https://sourceforge.net/projects/tgicl/files/">TGICL</a></td>
				<td> 2.2.0 </td>
				<td>  </td>
		  </tr>
		  <tr class="mod-list assembly">
				<td> <a target="_blank" href="https://github.com/trinityrnaseq/trinityrnaseq/wiki">Trinity</a></td>
				<td> 2.2.0 </td>
				<td>  </td>
		  </tr>
		  <!-- TOOLS post-mappings -->
		  <tr class="mod-list post-mappings">
				<td> <a target="_blank" href="https://www.broadinstitute.org/gatk/">GATK</a></td>
				<td> 3.3 </td>
				<td> gatkBaseRecalibrator <br/> gatkRealignerTargetCreator <br/> gatkIndelRealigner <br/> gatkHaplotypeCaller <br/> gatkSelectVariants <br/> gatkVariantFiltration <br/> gatkUnifiedGenotyper <br/> gatkBaseRecalibrator <br/> gatkPrintReads
		  </tr>
		  <tr class="mod-list post-mappings">
				<td> <a target="_blank" href="http://samtools.sourceforge.net/">SAMtools</a></td>
				<td> 0.1.3 </td>
				<td> samToolsFaidx <br/> samToolsIndex <br/> samToolsView <br/> samToolsSort <br/> mergeHeader <br/> samToolsMerge <br/> samToolsIdxstats <br/> samToolsDepth <br/> samToolsFlagstat <br/> samToolsMpileUp
		  </tr>
		  <tr class="mod-list post-mappings">
				<td> <a target="_blank" href="http://broadinstitute.github.io/picard/">PicardTools</a></td>
				<td> 1.63 </td>
				<td> picardToolsMarkDuplicates <br/> picardToolsCreateSequenceDictionary <br/> picardToolsSortSam <br/> picardToolsAddOrReplaceReadGroup <br/> picardToolsValidateSamFile <br/> picardToolsCleanSam <br/> picardToolsSamFormatConverter
		  </tr>
		  <tr class="mod-list post-mappings">
				<td> <a target="_blank" href="http://ngsutils.org/">NGSUtils</a></td>
				<td> 0.5.9</td>
				<td> bamutils </td>
		  </tr>
		  <!-- TOOLS vcf-tools -->
		  <tr class="mod-list vcf-tools">
				<td> <a target="_blank" href="http://snpeff.sourceforge.net/">Snpeff</a></td>
				<td> 4.2 </td>
				<td> snpeffAnnotation </td>
		  </tr>
		  <!-- TOOLS rna-seq -->
		  <tr class="mod-list rna-seq">
				<td> <a target="_blank" href="http://www-huber.embl.de/HTSeq/doc/count.html">HTSeq-Count</a></td>
				<td>  </td>
				<td>  </td>
		  </tr>
		  <!-- TOOLS struc-variants -->
		  <tr class="mod-list struc-variants">
			<td> <a target="_blank" href="http://breakdancer.sourceforge.net/">Breakdancer</a></td>
			<td> 1.4.5 </td>
			<td>  </td>
		  </tr>
		  <tr class="mod-list struc-variants">
				<td> <a target="_blank" href="http://gmt.genome.wustl.edu/packages/pindel/">Pindel</a></td>
				<td> 0.2.4 </td>
				<td>  </td>
		  </tr>
		  <!-- TOOLS optional -->
		  <tr class="mod-list optional">
				<td> <a target="_blank" href="http://www.graphviz.org/">Graphviz</a></td>
				<td> v2.xx </td>
				<td>  </td>
		  </tr>
		  </table>
	  </div>
	  <div class="modal-footer">
		<button type="button" class="btn btn-primary" data-dismiss="modal">Close</button>
	  </div>
	</div>
	</div>
</div>
